internal/webhooks/cluster.go

```diff
@@ -882,3 +885,68 @@ func validateTopologyMetadata(topology *clusterv1.Topology, fldPath *field.Path)
 	}
 	return allErrs
 }
+
+// validateAutoscalerAnnotationsForCluster iterates the MachineDeploymentsTopology objects under Workers and ensures the replicas
+// field and min/max annotations for autoscaler are not set at the same time. Optionally it also checks if a given ClusterClass has
+// the annotations that may apply to this Cluster.
+func validateAutoscalerAnnotationsForCluster(cluster *clusterv1.Cluster, clusterClass *clusterv1.ClusterClass) field.ErrorList {
+	var (
+		annotationMatches int
+		allErrs           field.ErrorList
+	)
+	if cluster.Spec.Topology == nil || cluster.Spec.Topology.Workers == nil {
+		return allErrs
+	}
+	fldPath := field.NewPath("spec", "topology")
+	for i, mdt := range cluster.Spec.Topology.Workers.MachineDeployments {
+		if mdt.Replicas == nil {
+			continue
+		}
+		annotationMatches = 0
+		for k := range mdt.Metadata.Annotations {
+			if k == clusterv1.AutoscalerMinSizeAnnotation || k == clusterv1.AutoscalerMaxSizeAnnotation {
+				annotationMatches++
+			}
+		}
+		if annotationMatches == 2 {
```

Comment:
Q: Should we already return an error as soon as 1 annotation is set? While the autoscaler doesn't become active then, it doesn't make much sense to allow folks to set replicas + 1 annotation

(this would also cover the strange case where 1 annotation is set in the CC and the other one on the Cluster)

Reply:
do we have to be cautious about a user who is updating the annotations but, due to timing or mistake, has an update of one annotation without the other?

Reply:
I think we expect users to make those changes within one single edit / update request

Reply:
> Q: Should we already return an error as soon as 1 annotation is set? 

IMO, no.

as we asked on slack - 2 annotations means autoscaler is active on this object, but not < 2.
https://kubernetes.slack.com/archives/C8TSNPY4T/p1712070139659279?thread_ts=1712064419.251189&cid=C8TSNPY4T

Reply:
Yeah I know. I just think having one annotation and replicas set is sla pretty confusing/ unclear state. And I don't see why we should allow it

Reply:
if autoscaller docs are clear to users that 2 annotations are required for the component to work, it's just a state where the autoscaler is not enabled and replicas are allowed. i think there is no use case for it, i.e. it seems like a user mistake to only have one annotation, and that's for the autoscaler to error / handle.

yet, @sbueringer @elmiko please +1 if you'd want to see the logic here to error on >= 1 annotations.

Reply:
I would prefer to already block on one annotation. This makes it clear to anyone looking at the cluster YAML what's going on without consulting the autoscaler docs.

But not a showstopper for me.

If we only want to block if both are set, please also cover the case where one annotation is set on the CC and the other one on the Cluster (because this also leads to a MD that has both)

Reply:
+1 for blocking already at 1 annotation, this helps to guide users to not make mistakes.

Reply:
+1 to block, allowing user to have "half autoscaler" seems a mistake (also the approach we are using is to be explicit)